src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -148,20 +150,18 @@ int BulkController::open() {
     }
 
     /* Look up endpoint addresses in supported database */
-    int i;
-    for (i = 0; bulk_supported[i].vendor_id; ++i) {
-        if ((bulk_supported[i].vendor_id == m_vendorId) &&
-                (bulk_supported[i].product_id == m_productId)) {
-            m_inEndpointAddr = bulk_supported[i].in_epaddr;
-            m_outEndpointAddr = bulk_supported[i].out_epaddr;
+    const bulk_supported_t* device = std::find(
+            std::begin(bulk_supported), std::end(bulk_supported), bulk_supported_t {
```

Comment:
Before only the to vendor and product ID have been compared, now this is a deep compare. Is there a reason? I suggest to restore the old faster lookup via a lambda. 

Regarding the code style, I think we should put the { without space to the type to match a constructor call style. Is there a way to put that into clang-format?

Reply:
> Before only the to vendor and product ID have been compared, now this is a deep compare. Is there a reason? I suggest to restore the old faster lookup via a lambda.

whops, I completely misread the code here, of course that makes more sense. 

> Regarding the code style, I think we should put the { without space to the type to match a constructor call style. Is there a way to put that into clang-format?

Yes, there is [SpaceBeforeCpp11BracedList](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#spacebeforecpp11bracedlist)

Reply:
fixed